### ISSUE TITLE
resize_image does not generate images when called from a template

### ISSUE THREAD

**Issue by callahad** [getzola/zola#669]
Zola version: 0.6.0

Steps to Reproduce:

1. Use `resize_image` inside a page or section template.

Observations:

1. No images are generated or saved to disk.
2. Filename hashes *are* calculated correctly
3. The same exact function invoked from a Markdown shortcode works as expected.

Minimal test site at https://github.com/callahad/zola-testcases/tree/master/resize-image-in-template

**Comment by Keats** [comment]
Anyone interested in looking at it before I do?

Thanks for the reproduction!

**Comment by Keats** [comment]
This should be fixed now in the `next` branch

**Comment by tforgione** [comment]
I had the problem previously, I tried the next branch on my website and I can confirm it works for me.

Thanks!

**EDIT** Actually it's quite weird. The first time I run `zola build`, the images are not generated, but the second time I run it, they are. This is what I get from running @callahad's example

``` sh
$  zola build # First build
Building site...
-> Creating 0 pages (0 orphan), 0 sections, and processing 1 images
Done in 34ms.
$ ls public # No processed images
404.html  index.html  robots.txt  sailboat.png  sitemap.xml
$ zola build # Second build
Building site...
-> Creating 0 pages (0 orphan), 0 sections, and processing 1 images
Done in 26ms.
$ ls public # Processed images are there
processed_images  404.html  index.html  robots.txt  sailboat.png  sitemap.xml
```

**Comment by Keats** [comment]
@tforgione I cannot reproduce the issue in the edit? I always get `processed_images` after `rm -rf public/ && zola build`

**Comment by tforgione** [comment]
Hum... I cannot reproduce it either..., maybe something weird about my config... or maybe I made some weird mistake. Sorry for the inconveniance!

**Comment by tforgione** [comment]
Ok, so after more attempts, I think I get what is happening.

If I clone a zola repository, let's say like this
```
git clone --recurse-submodules https://github.com/tforgione/zola-academic-example
```

and if I run `zola build` once, the processed images appear in `static/processed_images`. If I run `zola build` a second time, they also appear `public/processed_images`.

My hypothesis is that the processed images are generated into the `static/processed_images` directory, but after the `static` directory has been copied into the `public` directory.

If my hypothesis is correct, you should be able to reproduce the bug by removing `static/processed_images` and `public`, and then re-running `zola build`.

I hope this helps.

Thank you!

**Comment by Keats** [comment]
Ah yes that's obvious in hindsight. I need to copy the static dirs AFTER processing the images >_> I've pushed a commit doing that. Thanks a lot!

**Comment by tforgione** [comment]
Glad I could help! Tried the commit and it works fine for me! Thanks! :+1:

**Comment by Follpvosten** [comment]
I'm probably just doing something totally wrong here, but...
![image](https://user-images.githubusercontent.com/22855787/58935965-eb93e000-876e-11e9-910f-e21a001675b1.png)
This worked fine in 0.5.1, doesn't work at all in 0.7.0. The value of the `url` variable is the same as before, and I don't get any errors (unless I change the path to something invalid). There are *no* images generated whatsoever. Is this the same issue?

**Comment by tforgione** [comment]
@Follpvosten are you running the `next` branch ? The bug should be patched on this branch.

**Comment by Follpvosten** [comment]
No, this is the latest release (0.7.0). I was just curious because the behaviour is different on my side (no images get emitted at all).

**Comment by Keats** [comment]
That's the same as the parent issue I think, it should be fixed in the `next` branch (0.8.0) which isn't released yet
